xarray_beam/_src/core.py

```diff
@@ -307,13 +324,36 @@ def __init__(
     self.sharded_dim = max(lengths, key=lengths.get) if lengths else None
     self.shard_count = self._shard_count()
 
+  @property
+  def _first(self) -> xarray.Dataset:
+    return self._datasets[0]
+
+  @property
+  def _datasets(self) -> List[xarray.Dataset]:
+    if type(self.dataset) is xarray.Dataset:
+      return [self.dataset]
+    return list(self.dataset)
+
+  def _validate(self, dataset, split_vars):
+    """Raise errors if input parameters are invalid."""
+    if not dataset and not type(xarray.Dataset):
+      raise ValueError('dataset list cannot be empty!')
+    if not _all_equal(ds.sizes for ds in self._datasets):
+      raise ValueError('all datasets must be the same size')
```

Comment:
This is fine, but just for reference, I usually write these sort of checks by creating a list and then checking for equality (e.g., by checking of a `set` of size greater than 1), something like:
```python
sizes = [ds.sizes for ds in self._datasets]
if len(set(sizes)) > 1:
  raise ValueError(f'inconsistent dataset sizes: {sizes}')
```

The main advantage is that it results in better error messages.